evennia/contrib/wilderness.py

```diff
@@ -0,0 +1,723 @@
+    """
+
+    @property
+    def wilderness(self):
+        """
+        Shortcut property to the wilderness script.
+
+        Returns:
+            WildernessScript: the WildernessScript attached to this exit's room
+        """
+        return self.location.wilderness
+
+    @property
+    def mapprovider(self):
+        """
+        Shortcut property to the map provider.
+
+        Returns:
+            MapProvider object: the mapprovider object used with this
+                wilderness map.
+        """
+        return self.wilderness.mapprovider
+
+    def at_traverse_coordinates(self, traversing_object, current_coordinates,
+                                new_coordinates):
+        """
+        Called when an object wants to travel from one place inside the
+        wilderness to another place inside the wilderness.
+
+        If this returns True, then the traversing can happen. Otherwise it will
+        be blocked.
+
+        This method is similar how the `at_traverse` works on normal exits.
+
+        Args:
+            traversing_object (Object): The object doing the travelling.
+            current_coordinates (tuple): (x, y) coordinates where
+                `traversing_object` currently is.
+            new_coordinates (tuple): (x, y) coordinates of where
+                `traversing_object` wants to travel to.
+
+        Returns:
+            bool: True if traversing_object is allowed to traverse
+        """
+        return True
+
+    def at_traverse(self, traversing_object, target_location):
+        """
+        This implements the actual traversal. The traverse lock has
+        already been checked (in the Exit command) at this point.
+
+        Args:
+            traversing_object (Object): Object traversing us.
+            target_location (Object): Where target is going.
+
+        Returns:
+            bool: True if the traverse is allowed to happen
+
+        """
+        itemcoordinates = self.location.wilderness.db.itemcoordinates
+
+        current_coordinates = itemcoordinates[traversing_object]
+        new_coordinates = get_new_coordinates(current_coordinates, self.key)
+
+        if not self.at_traverse_coordinates(traversing_object,
+                                            current_coordinates,
+                                            new_coordinates):
+            return False
+
+        if not traversing_object.at_before_move(None):
+            return False
+        traversing_object.location.msg_contents("{} leaves to {}".format(
+            traversing_object.key, new_coordinates),
+            exclude=[traversing_object])
+
+        self.location.wilderness.move_obj(traversing_object, new_coordinates)
+
+        traversing_object.location.msg_contents("{} arrives from {}".format(
+            traversing_object.key, current_coordinates),
+            exclude=[traversing_object])
+
+        traversing_object.at_after_move(None)
+        return True
+
+
+class WildernessMapProvider(object):
```

Comment:
Since this is sort of a default input to the other entities, it should probably be at the very end, after the WildernessScript, which is not intended to be modified as far as I understand.